include/onnxruntime/core/framework/allocator.h

```diff
@@ -15,21 +15,78 @@
 #include "core/framework/fence.h"
 #include "core/session/onnxruntime_c_api.h"
 
+// Struct to represent a physical device.
+struct OrtDevice {
+  typedef int8_t DeviceType;
+  typedef int8_t MemoryType;
+  typedef int16_t DeviceId;
+
+  // Pre-defined device types.
+  static const DeviceType CPU = 0;
+  static const DeviceType GPU = 1; //CUDA
+  static const DeviceType FPGA = 2;
+
+  // Pre-defined memory types.
+  static const MemoryType Default = 0;
+  static const MemoryType CUDA_PINNED = 1;
+
```

Comment:
At the ORT framework level do we need to know the specifics (was created by CUDA and is 'pinned') or would something more generic like 'this is host accessible memory' be as good? Just wondering if the latter approach would make logic around whether a copy is possible simpler.